master/pkg/stream/stream.go

```diff
@@ -132,28 +146,32 @@ func NewSubscription[T Msg](
 func (s *Subscription[T]) Register() {
 	s.Publisher.Lock.Lock()
 	defer s.Publisher.Lock.Unlock()
-	s.Publisher.Subscriptions[s] = struct{}{}
+	s.element = s.Publisher.Subscriptions.PushBack(s)
 }
 
 // Unregister removes a Subscription from its Publisher.
 func (s *Subscription[T]) Unregister() {
 	s.Publisher.Lock.Lock()
 	defer s.Publisher.Lock.Unlock()
-	delete(s.Publisher.Subscriptions, s)
+	subscriptions := s.Publisher.Subscriptions
+	subscriptions.Remove(s.element)
 }
 
 // Publisher is responsible for publishing messages of type T
 // to streamers associate with active subscriptions.
 type Publisher[T Msg] struct {
 	Lock          sync.Mutex
-	Subscriptions map[*Subscription[T]]struct{}
+	Subscriptions *list.List
```

Comment:
I will change this back to map. The previous hydration(hydrate message while broadcasting) requires the knowledge of the subscription order for testing purposes. If the broadcast order to users changes, the output message will also change. This is no longer an issue.

Reply:
updated.